### ISSUE TITLE
TIMESTAMP function and ISO8601 issue

### ISSUE THREAD

**Issue by boie0025** [precog/quasar#1537]
We're sending in a string (`"2016-09-10T00:00:00-05:00"`) for TIMESTAMP to parse, but getting an error back:

`"Date/time string could not be parsed as timestamp: 2016-09-10T00:00:00-05:00"`

On Quasar 12.4.0

Here's a bit more of the error output:

```
RuntimeError - create_temp_collection QuasarRestClient.long_query call failed: 400 - { "error": { "status": "Bad Request", "detail": { "errors": [ { "status": "Malformed date/time string.", "detail": { "message": "Date/time string could not be parsed as timestamp: 2016-09-10T00:00:00-05:00", "functionName": "timestamp", "input": "2016-09-10T00:00:00-05:00" } } ],
```

From my reading, it looks like I am supposed to send in an ISO8601 string to TIMESTAMP, am I missing something, or is there a bug?

**Comment by damonLL** [comment]
@boie0025 Looks like the format might be incorrect.  The docs do need updating, but the date/time section should be valid: http://quasar-analytics.org/docs/sqlreference/#dates-and-times

Format should be `2015-04-29T15:16:55`

**Comment by damonLL** [comment]
@boie0025 Should be something like `("2015-04-29T15:16:55")` when querying

**Comment by boie0025** [comment]
@damonLL Thanks for the response!

I tried the timestamp you gave me directly against Quasar:
```SQL
SELECT TIMESTAMP("2015-04-29T15:16:55") AS time;
```

and it failed with `"Date/time string could not be parsed as timestamp: 2015-04-29T15:16:55"`

but when I tried

```SQL
SELECT TIMESTAMP("2015-04-29T15:16:55Z") as time;
```

It ran as expected.  Looking through the code for the timestamp function I found this which is a regex that makes the Z optional: https://github.com/quasar-analytics/quasar/blob/575aff353433ea21d59dbcdcaae8cca53b042d05/frontend/src/main/scala/quasar/std/string.scala#L258

But I'm not sure if that's related or not?  I do know that we originally went from the docs, and found that the Z was always required, so I tried using other ISO8601 strings expecting them to work, but no dice.  We can add the Z, and convert all times to UTC/Zulu for now, but let me know what you find.

**Comment by zeeiyerWork** [comment]
Looks like there is a workaround - confirm if this is a bug or not.

**Comment by edmundnoble** [comment]
The problem in the original issue is that there's no parsing for timestamps with timezones attached. That will be fixed along with #2564. In the following comment however, there is an issue which we can't resolve in a way which makes sense; "2015-04-29T15:16:55" is not a timestamp. Timestamps are fixed moments in time, this is not grounded in any timezone (even UTC) so even after #2564 is resolved it's not going to be parsed as a timestamp, just a combination of a date and a time, which is a useful notion on its own. You will still need `Z` to denote that a timestamp is in UTC.
